tests/src/it/scala/cr/pulsar/PulsarSpec.scala

```diff
@@ -78,95 +81,120 @@ class PulsarSpec extends PulsarSuite {
       }
     }
 
-    test("A consumer fails to decode a message") {
-      val dfTopic = topic("decoding-failure")
+    test("Schema versioning: producer sends old Event, Consumer expects Event_V2") {
```

Comment:
What would happen if there would be a `Producer` emitting `Event_V2` and a `Consumer` expecting `Event`?

Reply:
It will work if the `Event` decoder knows how to read `Event_V2` in the same way `Event_V2` can process `Event`. To properly test schema versioning we need to use the same class (see the `ClassTag` requirement) for which we'll need to deploy, for example, different branches of the same code but I requires much more time working on this and I already got Jack giving me other priorities so this'll need to wait.